docs/cross-account-resource-management.md

```diff
@@ -0,0 +1,56 @@
+# Cross account resource management (CARM)
+
+## The problem
+
+Today adminstrative users can manage AWS resources using ACK. However, these resources are created in the same AWS account as the Kubernetes cluster. This leads to a suboptimal experience for users in large organisations: creating resources in multiple AWS accounts requires ACK to be installed and managed in many kubernetes clusters. Considering ACK has dozens of service controllers, this means administrators must manage hundreds of AWS service controller pods across many kubernetes clusters and many AWS Accounts with scoped roles.
+
+With cross account resource management, a single kubernetes cluster with ACK installed can manage the resources of many different AWS accounts. Kubernetes administrators will not need to install and manage multiple clusters and ACK controllers, and AWS administrators will spend less effort managing permissions to access the services across accounts.
+
+## Proposed implementation
+
+The proposed solution will solve the problem of cross account management within a single kubernetes cluster. Given an AWS Owner account ID it will look up an IAM Role ARN and call STS:AssumeRole to pivot the AWS SDK calls in order to execute in a new AWS Account and role context.
+
+Controllers will store AccountIDs and their associated IAM Role ARNs in a ConfigMap or Secret Kubernetes object. Cluster administrators will be responsible for administrating and updating the configmap/secret, and creating the target roles using the AWS Console/CLI. Controllers can reuse and existing IAM Role ARN if it matches the scope they need. 
+
+![carm-flowchart](./images/carm-flowchart.png)
+
+## Potential other solutions
+
+- Store the IAM Role ARNs and account IDs as a kubernetes objects (new CRDs)
+- Hard coded configuration loaded at the start time (need to restart the controller to update the configuration)
+- Specify the role using an annotation (painful user experience, security problems)
+
+## In scope
+
+- Ability to have the service controller assume the same IAM role in a different account.
+- Tracebility and audibility of IAM Role related events
+- User has ability to specify the AWS account in CR Annotations
+- Unit and e2e tests with a specific ACK Controller (e.g. s3 bucket controller)
+
+## Out of scope
+
+- Any service controller specific code (e.g generated code)
+- CARM e2e tests for all controllers
+- Determination of owner account ID
```

Comment:
right. an Owner Account ID will be already determined and passed to the `NewSession` function:

https://github.com/aws/aws-service-operator-k8s/blob/717d4ee368c4aec2b9ead5d11f4b6e89b3d082e6/pkg/runtime/session.go#L18